### ISSUE TITLE
Update enrollment codes

### ISSUE THREAD

**Issue by SritterCL** [CarnegieLearningWeb/educational-experiment-service#103]
It looks like the enrollment codes are strings, rather than numbers (enums would be OK).

It also looks like enrollment code was set to null on an error (rather than an error code). Here's my suggestion:
0 - participant was included in experiment
1 - participant excluded due to unspecified error
2 - participant reached experiment point prior to experiment enrolling
3 - participant was on exclusion list
4 - participant’s group was on exclusion list
5 - participant excluded due to group assignment logic
6 - participant excluded from group-assignment experiment - no group specified

(we should try not to use #1 - any "common" errors should get their own code).

**Comment by jerith666** [comment]
I'd definitely vote for an enum over a number.

**Comment by nirmalpatel** [comment]
@jerith666 do you mean that rather than the numbers, there would be short strings like these:

INCLUDE - participant was included in experiment
ERROR - participant excluded due to unspecified error
REACHED_PRIOR - participant reached experiment point prior to experiment enrolling
...

?

**Comment by jerith666** [comment]
Yes.  And those strings should be the values of a TypeScript enum.

**Comment by SritterCL** [comment]
I'm fine with using strings like that. How about:

INCLUDE - participant was included in experiment
ERROR - participant excluded due to unspecified error
REACHED_PRIOR - participant reached experiment point prior to experiment enrolling
PARTICIPANT_ON_EXCLUSION_LIST - participant was on exclusion list
GROUP_ON_EXCLUSION_LIST - participant’s group was on exclusion list
EXCLUDED_DUE_TO_GROUP - participant excluded due to group assignment logic
NO_GROUP_SPECIFIED - participant excluded from group-assignment experiment - no group specified

**Comment by VivekFitkariwala** [comment]
Should we use Enrollment Codes or Assignment Codes?

**Comment by SritterCL** [comment]
Just to check my understanding, assignment is about getAllExperimentConditions and enrollment is about markExperiment Point, right?

I think these apply to both enrollment and assignment.
